compiler/aarch64/codegen/OMRTreeEvaluator.cpp

```diff
@@ -2479,10 +2479,93 @@ inlineVectorMaskedBinaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode
    return resReg;
    }
 
+/**
+ * @brief Helper functions for generating instruction sequence for masked unary operations
+ *
+ * @param[in] node: node
+ * @param[in] cg: CodeGenerator
+ * @param[in] op: unary opcode
+ * @return vector register containing the result
+ */
+static TR::Register *
+inlineVectorMaskedUnaryOp(TR::Node *node, TR::CodeGenerator *cg, TR::InstOpCode::Mnemonic op)
+   {
+   TR::Node *firstChild = node->getFirstChild();
+   TR::Node *secondChild = node->getSecondChild();
+
+   TR::Register *srcReg = cg->evaluate(firstChild);
+
+   TR_ASSERT_FATAL_WITH_NODE(node, srcReg->getKind() == TR_VRF, "unexpected Register kind");
+
+   TR::Register *resReg = cg->allocateRegister(TR_VRF);
+   node->setRegister(resReg);
+
+   generateTrg1Src1Instruction(cg, op, node, resReg, srcReg);
+
+   TR::ILOpCode secondOp = secondChild->getOpCode();
+   bool flipMask = false;
+   TR::Register *maskReg = NULL;
+   VectorCompareOps compareOp;
+   if (secondOp.isVectorOpCode() && secondOp.isBooleanCompare() && (!secondOp.isVectorMasked())
+       && ((compareOp = getVectorCompareOp(secondOp.getVectorOperation())) != VECTOR_COMPARE_INVALID)
+       && (secondChild->getReferenceCount() == 1) && (secondChild->getRegister() == NULL))
+      {
+      maskReg = vcmpHelper(secondChild, compareOp, true, &flipMask, cg);
+      }
+   else
+      {
+      maskReg = cg->evaluate(secondChild);
+      }
+   TR_ASSERT_FATAL_WITH_NODE(node, maskReg->getKind() == TR_VRF, "unexpected Register kind");
+
+   if (flipMask)
+      {
+      /* BIT inserts each bit from the first source if the corresponding bit of the second source is 1. */
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::vbit16b, node, resReg, srcReg, maskReg);
+      }
+   else
+      {
+      /* BIF inserts each bit from the first source if the corresponding bit of the second source is 0. */
+      generateTrg1Src2Instruction(cg, TR::InstOpCode::vbif16b, node, resReg, srcReg, maskReg);
+      }
```

Comment:
I would suggest introducing a new variable for the bitwise insert operation like `insertOp = flipMask ? TR::InstOpCode::vbit16b : TR::InstOpCode::vbif16b;`

Reply:
Yes, that would change if-else branch into a ternary operator and make the function more concise. 
I pushed a new commit to use the ternary operator directly in the invocation of generateTrg1Src2Instruction instead of defining a new variable. What do you think?

Reply:
Thank you.
Putting the ternary expression in `generateTrg1Src2Instruction()` is fine.  It is readable enough.